src/game.cpp

```diff
@@ -4857,14 +4857,30 @@ void Game::kickPlayer(uint32_t playerId, bool displayEffect)
 	player->kickPlayer(displayEffect);
 }
 
+void Game::playerReportRuleViolationReport(uint32_t playerId, const std::string& targetName, uint8_t reportType, uint8_t reportReason, const std::string& comment, const std::string& translation)
+{
+	Player* player = getPlayerByID(playerId);
+	if (!player) {
+		return;
+	}
+	
+	Player* target = getPlayerByName(targetName);
+	if (!target) {
+		// NOTE: target has logged out, maybe just log?
```

Comment:
@ranisalt here, should i send userdata of target for lua or string with player name? player may evade a report by logging out for example.